libres/lib/include/ert/res_util/memory.hpp

```diff
@@ -0,0 +1,59 @@
+/*
+   Copyright (C) 2011  Equinor ASA, Norway.
+
+   The file 'memory.hpp' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+#include <ert/python.hpp>
+
+#ifndef ERT_MEMORY_H
+#define ERT_MEMORY_H
+
+long SystemRamFree(void);
+long ProcessMemory(void);
+long ProcessMaxMemory(void);
+long ProcessMaxRSS(void);
+
+class ScopedMemoryLogger {
+public:
+    ScopedMemoryLogger(std::shared_ptr<ert::ILogger> logger,
+                       const std::string message)
+        : m_logger(logger), m_message(message) {
+        m_enter_mem = ProcessMemory();
```

Comment:
The `m_enter_mem` and `m_enter_max_mem` can also be initialized in the initializer list:
```
      : m_logger(logger)
      , m_message(message)
      , m_enter_mem(ProcessMemory())
      , m_enter_max_mem(ProcessMaxMemory)
      {
            this->m_logger->info( .... );
      }
```